content/ja/docs/tutorials/services/source-ip.md

```diff
@@ -0,0 +1,421 @@
+the target localization.
+{{< /comment >}}
+
+[NAT](https://ja.wikipedia.org/wiki/%E3%83%8D%E3%83%83%E3%83%88%E3%83%AF%E3%83%BC%E3%82%AF%E3%82%A2%E3%83%89%E3%83%AC%E3%82%B9%E5%A4%89%E6%8F%9B)
+: ネットワークアドレス変換(network address translation)
+
+[送信元NAT](https://en.wikipedia.org/wiki/Network_address_translation#SNAT)
+: パケットの送信元のIPを置換します。このページでは、通常ノードのIPアドレスを置換することを意味します。
+
+[送信先NAT](https://en.wikipedia.org/wiki/Network_address_translation#DNAT)
+: パケットの送信先のIPを置換します。このページでは、通常{{< glossary_tooltip term_id="pod" >}}のIPアドレスを置換することを意味します。
+
+[VIP](/ja/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies)
+: Kubernetes内のすべての{{< glossary_tooltip text="Service" term_id="service" >}}などに割り当てられる仮想IPアドレス(virtual IP address)です。
+
+[kube-proxy](/ja/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies)
+: すべてのノード上でServiceのVIPを管理するネットワークデーモンです。
+
+### 前提条件
+
+{{< include "task-tutorial-prereqs.md" >}}
+
+以下の例では、HTTPヘッダー経由で受け取ったリクエストの送信元IPをエコーバックする、小さなnginxウェブサーバーを使用します。次のコマンドでウェブサーバーを作成できます。
+
+```shell
+kubectl create deployment source-ip-app --image=k8s.gcr.io/echoserver:1.4
+```
+
+出力は次のようになります。
+
+```
+deployment.apps/source-ip-app created
+```
+
+## {{% heading "objectives" %}}
+
+* 単純なアプリケーションを様々な種類のService経由で公開する
+* それぞれの種類のServiceがどのように送信元IPのNATを扱うかを理解する
+* 送信元IPを保持することに関わるトレードオフを理解する
+
+<!-- lessoncontent -->
+
+## `Type=ClusterIP`を使用したServiceでの送信元IP
+
+kube-proxyが[iptablesモード](/ja/docs/concepts/services-networking/service/#proxy-mode-iptables)(デフォルト)で実行されている場合、クラスター内部からClusterIPに送られたパケットに送信元のNATが行われることは決してありません。kube-proxyが実行されているノード上で`http://localhost:10249/proxyMode`にリクエストを送って、kube-proxyのモードを問い合わせてみましょう。
+
+```console
+kubectl get nodes
+```
+
+出力は次のようになります。
+
+```
+NAME                           STATUS     ROLES    AGE     VERSION
+kubernetes-node-6jst   Ready      <none>   2h      v1.13.0
+kubernetes-node-cx31   Ready      <none>   2h      v1.13.0
+kubernetes-node-jj1t   Ready      <none>   2h      v1.13.0
+```
+
+これらのノードの1つでproxyモードを取得します(kube-proxyはポート10249をlistenしています)。
+
+```shell
+# このコマンドは、問い合わせを行いたいノード上のシェルで実行してください。
+curl http://localhost:10249/proxyMode
+```
+
+出力は次のようになります。
+
+```
+iptables
+```
+
+source IPアプリのServiceを作成することで、送信元IPが保持されているかテストできます。
+
+```shell
+kubectl expose deployment source-ip-app --name=clusterip --port=80 --target-port=8080
+```
+
+出力は次のようになります。
+
+```
+service/clusterip exposed
+```
+```shell
+kubectl get svc clusterip
+```
+
+出力は次のようになります。
+
+```
+NAME         TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)   AGE
+clusterip    ClusterIP   10.0.170.92   <none>        80/TCP    51s
+```
+
+そして、同じクラスター上のPodから`ClusterIP`にアクセスします。
+
+```shell
+kubectl run busybox -it --image=busybox --restart=Never --rm
+```
+
+出力は次のようになります。
+
+```
+Waiting for pod default/busybox to be running, status is Pending, pod ready: false
+If you don't see a command prompt, try pressing enter.
+
+```
+
+これで、Podの内部でコマンドが実行できます。
+
+```shell
+# Run this inside the terminal from "kubectl run"
```

Comment:
If you want, you can translate the comment.

Reply:
Oh, I missed it. Thank you.

Reply:
Fixed it!